diff --git a/Sources/krafix.cpp b/Sources/krafix.cpp
index 990b440..fe1dc33 100644
--- a/Sources/krafix.cpp
+++ b/Sources/krafix.cpp
@@ -1171,7 +1171,7 @@ int compileWithTextureUnits(const char* targetlang, const char* from, std::strin
 	return errors;
 }
 
-void krafix_compile(const char* source, char* output, int* length, const char* targetlang, const char* system, const char* shadertype) {
+extern "C" void krafix_compile(const char* source, char* output, int* length, const char* targetlang, const char* system, const char* shadertype) {
 	std::string defines;
 	std::vector<int> textureUnitCounts;
 	bool instancedoptional = false;
diff --git a/kincfile.js b/kincfile.js
index 719fafd..ff5da63 100644
--- a/kincfile.js
+++ b/kincfile.js
@@ -1,6 +1,6 @@
 let project = new Project('krafix');
 
-let library = false;
+let library = true;
 
 project.addDefine('SPIRV_CROSS_KRAFIX');
 project.addDefine('ENABLE_HLSL');
